README.md

```diff
@@ -1,7 +1,298 @@
+  .fromProject(() => Project.fromDir(dirname(require.resolve('../test-app/package.json')), { linkDevDeps: true })))
+
+  // Define multiple scenarios, each scenario is derived from the template and has unique dependencies installed
+  .expand({
+    modern: async (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package' });
+    },
+
+    legacy: async function (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package--legacy' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package--legacy' });
+    }
+  })
+
+  // Emit derivative projects to the filesystem
+  .forEachScenario(scenario => {
+
+    // Define your test suite
+    module(scenario.name, function (hooks) {
+      hooks.beforeEach(assert => {
+        // ...
+      });
+
+      test('output directories exist', async function () {
+        // ...
+      });
+
+      test('package.json is modified appropriately', async function () {
+        /// ...
+      });
+    });
+  });
+```
+
+Note: no new project codebases will be emitted to your filesystem until `forEachScenario` is executed. This lets you manipulate scenarios before emitting their codebases.
+
+## Creating and manipulating project codebases
+
+Instead of defining a template project from disk as shown above, you can do it programmatically.
+
+`scenario-tester` relies on [stefanpenner/node-fixturify-project](https://github.com/stefanpenner/node-fixturify-project) (npm package [fixturify-project](https://www.npmjs.com/package/fixturify-project)).
+
+`fixturify-project` converts codebases between a filesystem representation and a JSON representation. For example, it lets you define a project (a test app or an npm package) as JSON:
+
+```js
+const { Project } = require('fixturify-project');
+
+const project = new Project('rsvp', '3.1.4', {
+  files: {
+    'filename1.js': 'module.exports = "Hello, World!"',
+    dirname1: {
+      'filename2.json': `{
+        "This is a": "multiline string",
+        "that represents": "a JSON file in your project",
+      }`,
+    },
+  },
+});
+```
+
+Regardless of whether you created a project from disk or JSON, it will be represented in a `Project` class instance from `fixturify-project`.
+
+You can emit your project codebase to filesystem:
+
+```js
+project.basedir = '/tmp/my-project'; // Optional. When not set, a random tmp path will be assigned to `basedir`.
+
+await project.write()
+```
+
+This produces the following structure on disk:
+
+```sh
+/tmp/my-project/filename1.js
+/tmp/my-project/dirname1/filename2.js
+```
+
+You can add dependencies to a project:
+
+```js
+project.addDependency('mocha', '5.2.0');
+project.addDependency('chai', '5.2.0');
```

Comment:
if you want to show addDependency here use fake/mock package names. These dependencies that are added have nothing to do with npm, they are created as fixturify-project Projects with a minimal setup to be considered an npm package and then added to your project

see https://github.com/embroider-build/embroider/blob/main/tests/scenarios/compat-addon-import-test.ts#L10-L32 for inspiration